src/components/Bookform.js

```diff
@@ -1,14 +1,42 @@
+import { useDispatch } from 'react-redux';
+import { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
 import { addBook } from '../redux/books/books';
 
-const Bookform = () => (
-  <>
-    <h4> ADD NEW BOOK</h4>
-    <form onSubmit={addBook} className="form-container">
-      <input type="text" name="name" placeholder="Book title" />
-      <input type="text" name="name" placeholder="Author" />
-      <input type="submit" value="Add Book" />
-    </form>
-  </>
-);
+const Bookform = () => {
+  const [value, setState] = useState({
+    title: '',
+    author: '',
+  });
+  const dispatch = useDispatch();
+
+  const inputHandler = (e) => {
+    const { name, value } = e.target;
+    setState((preData) => ({
+      ...preData,
+      [name]: value,
+    }));
+  };
+
+  const clickHandler = (e) => {
+    e.preventDefault();
+    const newBook = {
+      id: uuidv4(),
+      ...value,
+    };
+    dispatch(addBook(newBook));
+    setState({ title: '', author: '' });
+  };
+  return (
+    <>
+      <h4> ADD NEW BOOK</h4>
+      <form onSubmit={clickHandler} className="form-container">
```

Comment:
Optional.
Upon adding a book, you can consider resetting the values of the input field. Doing so will greatly enhance the user experience as the user doesn't have to clean the input field before adding another entry. 
![image](https://user-images.githubusercontent.com/33268927/191764760-6b3dbdce-220a-424e-b9d1-0743168c3520.png)